grammar.js

```diff
@@ -858,66 +854,67 @@ module.exports = grammar(C, {
     compound_literal_expression: ($, original) => choice(
       original,
       seq(
-        field('type', choice(
-          $._type_identifier,
-          $.template_type,
-          $.scoped_type_identifier
-        )),
+        field('type', $._class_name),
         field('value', $.initializer_list)
       )
     ),
 
-    scoped_field_identifier: $ => prec(1, seq(
+    dependent_identifier: $ => seq('template', $.template_function),
+    dependent_field_identifier: $ => seq('template', $.template_method),
+    dependent_type_identifier: $ => seq('template', $.template_type),
+
+    _scoped_namespace: $=> prec(1, seq(
       field('namespace', optional(choice(
```

Comment:
Since this is a breaking change anyway, I'm leaning toward renaming this field 'scope' since they are not always namespaces. The parser can't actually know the semantic type, but it may not necessarily be a namespace. It could be a type, template type, dependent name, etc.

Reply:
Makes sense.